### ISSUE TITLE
[HOLD for payment 2023-06-13] [$1000] Show assigned user as mention in the task preview

### ISSUE THREAD

**Issue by thienlnam** [Expensify/App#19089]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Create a task and select an assignee
2. Navigate to the assigned chat

## Expected Result:
The task preview should show the assignee in a "mention" like view
<img width="337" alt="Screenshot 2023-05-16 at 4 02 58 PM" src="https://github.com/Expensify/App/assets/30609178/14053597-e15c-4b2f-a659-873bcbf865e9">

## Actual Result:
Just shows the task title
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?
N/A
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013df24eedfe4e5c4e</li>
        <li>Upwork Job ID: 1658609112262893568</li>
        <li>Last Price Increase: 2023-05-16</li>
    </ul>
</details>

**Comment by phuchoang23** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Show assigned user as mention in the task preview

### What is the root cause of that problem?
Feature request

### What changes do you think we should make in order to solve the problem?
We need to get assignee email from `props.action.taskAssignee` and put it in this before task's title in this line https://github.com/Expensify/App/blob/811c0f36fba4953af008e6643349d95c949627e4/src/components/ReportActionItem/TaskPreview.js#L81

To style it as message mention. I think we can either:
- Make a copy of component [MentionUserRenderer](https://github.com/Expensify/App/blob/811c0f36fba4953af008e6643349d95c949627e4/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.js#L29)
- Or we can split component [MentionUserRenderer](https://github.com/Expensify/App/blob/811c0f36fba4953af008e6643349d95c949627e4/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.js#L29) so that we can reuse component presentation between MentionUserRenderer and TaskPreview or other place that need to render mentioned user.

**Comment by dukenv0307** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Show assigned user as mention in the task preview
### What is the root cause of that problem?
Feature request
### What changes do you think we should make in order to solve the problem?
Actually message in task preview is only has `taskTitle` is wrapper in text
https://github.com/Expensify/App/blob/811c0f36fba4953af008e6643349d95c949627e4/src/components/ReportActionItem/TaskPreview.js#L81
To show assigned user as mention in the task preview, we could create an html same as normal comment like this
```
const htmlForTaskPreview = `<comment><mention-user>@${props.action.taskAssignee}</mention-user> ${taskTitle}</comment>`;
```
And then re-use `RenderHTML` component to render it in task preview
```
<RenderHTML html={htmlForTaskPreview} />
```
### What alternative solutions did you explore? (Optional)

**Comment by kevinksullivan** [comment]
Unassigning as I am going OOO for several days

**Comment by sobitneupane** [comment]
Thanks for the proposals.

@dukenv0307 Looks like `props.action.taskAssignee` is not always defined. For instance, Create a new task with different person as assignee and Share Somewhere.
<img width="372" alt="Screenshot 2023-05-18 at 13 49 05" src="https://github.com/Expensify/App/assets/25876548/565bd9bd-b86a-41c6-9579-395f2baefafd">

<img width="535" alt="Screenshot 2023-05-18 at 13 43 30" src="https://github.com/Expensify/App/assets/25876548/917f1b2b-1fc2-4acd-9ed8-9033b78c89ae">

**Comment by dukenv0307** [comment]
@sobitneupane Thanks for instance, I dont't know why props.action is different between assignee and share somewhere. So to work for all case we can use `props.action.assignee || props.taskReport.managerEmail`.

**Comment by phuchoang23** [comment]
@sobitneupane I think we can use `props.taskReport.participants` to show assignee list.

**Comment by sobitneupane** [comment]
I am getting some weird issue. Sometimes even task title is missing in task report and is show as undefined.

<img width="484" alt="Screenshot 2023-05-18 at 19 00 14" src="https://github.com/Expensify/App/assets/25876548/d3d0f10e-0338-4cbd-8b72-2e6203480f48">

**Comment by dukenv0307** [comment]
@sobitneupane Above case seems to be an other bug in here https://github.com/Expensify/App/issues/19090

**Comment by sobitneupane** [comment]
Oh yes. Thanks @dukenv0307.

[Proposal](https://github.com/Expensify/App/issues/19089#issuecomment-1550654651) from @dukenv0307 looks good to me.

🎀👀🎀 C+ reviewed

cc: @bondydaa

**Comment by bondydaa** [comment]
@thienlnam if `props.action.assignee` isn't set then no one has been assigned yet right? so then we shouldn't show a mention on the task?

**Comment by bondydaa** [comment]
assigning since we can get this mostly implemented while we wait for confirmation from Jack but I believe tasks can be created without assignees and in those cases we shouldn't show any mention so I think we can probably make it so that we only render the mention when `props.action.assignee` is set.

**Comment by thienlnam** [comment]
Good call - there's a couple back-end change that needs to come with this, so let's put this on hold until those are deployed.

As people in the thread have mentioned, it's confusing to check taskAssignee / assignee so I'm going to consolidate it so we are returning the managerEmail everytime, and fixing [this](https://github.com/Expensify/App/issues/19090) bug should make sure that data is always available

**Comment by thienlnam** [comment]
@dukenv0307 We won't start the timer yet until the back-end changes are live and this is off hold so no worries about that

**Comment by bondydaa** [comment]
moving to weekly for now until the backend changes are done then

**Comment by thienlnam** [comment]
Okay, the back-end changes will likely go live within the next day. For some info - if we're fetching reportAction data and it is linked to a taskReport then it will contain the following:

childReportName: Title of the task
childManagerEmail: Assignee email of the task

If you're getting data of the task report itself, it will contain the following
reportName: Title of the task
managerEmail: Assignee email

Let me know if there are more questions with this - will update again when the back-end changes are live. We'll have a short timeframe when task titles are broken so it would be good to move quick with this

**Comment by trjExpensify** [comment]
👋 @thienlnam can you make sure that when we put an issue on hold, we state the issue number it's held on for ease of following the breadcrumbs. i.e `[Hold Web-E #1234]` Thanks!

**Comment by thienlnam** [comment]
For sure https://github.com/Expensify/Web-Expensify/pull/37580, updated!

**Comment by thienlnam** [comment]
> taskAssignee: Assignee email or phone number of the task
> taskTitle: title of the task

Is this for new tasks or existing tasks? With all the changes you should not be seeing these on the reportAction

Also for the task report those are correct - check this thread here https://github.com/Expensify/App/issues/19089#issuecomment-1560304253

**Comment by thienlnam** [comment]
I've linked a couple issues that should also be solved by this.

Basically we should check the task report details first, and if those don't exist then fall back using the reportAction data (the report task should have the latest data)

**Comment by bondydaa** [comment]
pr was merged so dropping this to weekly

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

Checklist won't be required for this issue as we are not solving a bug here.

**Comment by thienlnam** [comment]
Noting there was a regression added from this PR https://github.com/Expensify/App/issues/20539

**Comment by trjExpensify** [comment]
Okay, following the history I don't think this qualifies for the #efficiency bonus: 

- we took this off hold on May 30th, so the count starts from then instead of when the issue was assigned - 4:17pm BST 
- Merged on June 5th - 8.20pm BST
- Despite June 3rd & June 4th being the weekend, that's still not within 3 business days of the count (which would have been Friday June 2nd at 4:17pm BST. 

And with the regression here, that makes it:

- @dukenv0307 - $1000 for the fix (we are in the process of implementing equal penalty for regressions, but that hasn't been introduced or announced just yet, so I'm not applying it here). 
- @sobitneupane - $500 for C+
- @thienlnam - $0 for the bug report 😛 

Does that sound right to everyone?

**Comment by trjExpensify** [comment]
@dukenv0307 can I get your upwork profile please?

**Comment by trjExpensify** [comment]
Thanks, sent you an offer! :)

**Comment by trjExpensify** [comment]
Settled up with you both!
